src/generate-elements.ts

```diff
@@ -17,28 +17,50 @@ import * as jsonschema from 'jsonschema';
 import * as pathLib from 'path';
 
 import {Attribute, Element, ElementLike, ElementMixin, Elements, Property, SourceRange} from './elements-format';
+import {Document} from './model/document';
+import {Feature} from './model/feature';
 import {Attribute as ResolvedAttribute, Element as ResolvedElement, ElementMixin as ResolvedMixin, Property as ResolvedProperty, SourceRange as ResolvedSourceRange} from './model/model';
+import {Package} from './model/package';
 
 export type ElementOrMixin = ResolvedElement | ResolvedMixin;
 
 export function generateElementMetadata(
-    features: ElementOrMixin[], packagePath: string): Elements {
-  const elements =
-      features.filter((f) => f.kinds.has('element')) as ResolvedElement[];
-  const mixins =
-      features.filter((f) => f.kinds.has('element-mixin')) as ResolvedMixin[];
+    input: Package|Document[]|Feature[], packagePath: string): Elements {
+  let elements: Set<ResolvedElement>;
+  let mixins: Set<ResolvedMixin>;
+
+  if (input instanceof Array) {
+    const features = input as Array<Feature>;
+    const onlyDocuments = features.every((i) => i.kinds.has('document'));
```

Comment:
Arg, get unlucky in your features and end up with different behavior. This code was written before we had Document worked out. I say drop the option to pass in Features, and take `Package|Document[]` instead.